src/openms/source/ANALYSIS/OPENSWATH/OpenSwathScoring.cpp

```diff
@@ -46,13 +49,55 @@
 
 // basic file operations
 
+namespace OpenMS
+{
+  void sortSpectrumByMZ(OpenSwath::SpectrumPtr spec)
```

Comment:
IMHO passing a smart pointer is overkill here.
It should be sufficient to pass `OpenSwath::Spectrum& spec`, right?

Reply:
well, this is how we handle all access to spectra internally and it makes the data handling and the interfaces easier.

Reply:
what is the advantage of this unnecessary indirection? 
I can only see disadvantages:
 - worse performance (passing a shared pointer by copy, which means atomic count increase / lock)
 - user needs to known about the SharedPtr, when everything he really needs is the thing pointed to

This is a bad interface choice and its being discussed in various lectures as "don't do it".